src/modules/sentry/sentry.service.ts

```diff
@@ -29,7 +30,28 @@ export class SentryService implements OnModuleDestroy, OnApplicationShutdown {
   constructor(protected readonly configService: ConfigService) {}
 
   private isExcludedExceptions(exception: unknown): boolean {
-    return this.EXCLUDED_EXCEPTIONS.some((item) => exception instanceof item);
+    if (this.EXCLUDED_EXCEPTIONS.some((item) => exception instanceof item)) {
+      return true;
+    }
+
+    if (
+      exception instanceof QueryFailedError &&
+      exception.message.search(
+        'duplicate key value violates unique constraint'
+      ) !== -1
+    ) {
+      return true;
+    }
```

Comment:
I think we should include a comment explaining why we are excluding this error.
Please add/correct my explanation based on your understanding
```suggestion
    // We are excluding this exception from Sentry logs because this exception is expected behaviour from DB transaction validations
    // We have noticed that it is usually occurring when...
    if (
      exception instanceof QueryFailedError &&
      exception.message.search(
        'duplicate key value violates unique constraint'
      ) !== -1
    ) {
      return true;
    }
```